include/dap/session.h

```diff
@@ -199,6 +199,16 @@ class Session {
                     const std::shared_ptr<Writer>&) = 0;
   inline void bind(const std::shared_ptr<ReaderWriter>&);
 
+
+  // Alternative to bind() for control over which thread the request is processed.
+  // If bindNoThread is used, the user must call OnDataAvailable() whenever data is
```

Comment:
As `bindNoThread()` is doing part of the `bind()` call, I think it would be cleaner to split `bind()` up into two methods:

```c++

  // connect() connects this Session to an endpoint.
  // connect() can only be called once. Repeated calls will raise an error, but
  // otherwise will do nothing.
  virtual void connect(const std::shared_ptr<Reader>&,
                    const std::shared_ptr<Writer>&) = 0;
  inline void connect(const std::shared_ptr<ReaderWriter>&);

  // startProcessingMessages() starts a new thread to receive and dispatch incoming messages.
  virtual void startProcessingMessages() = 0;

  // bind() connects this Session to an endpoint using connect(), and then
  // starts processing incoming messages with startProcessingMessages().
  inline void bind(const std::shared_ptr<Reader>&,
                    const std::shared_ptr<Writer>&);
  inline void bind(const std::shared_ptr<ReaderWriter>&);

  // getPayload() blocks until the next incoming message is received, returning the
  // payload or an empty function if the connection was lost.
  // The returned payload is function that can be called on any thread to dispatch the
  // message to the Session handler.
  virtual std::function<void()> getPayload() = 0;
```

Reply:
Updated!